R/spaces.R

```diff
@@ -165,11 +165,276 @@ space_create <- function(name, spaces_key = NULL, spaces_secret = NULL, ...) {
   spaces_secret <- check_space_secret(spaces_secret)
 
   res <- put_bucket(name,
-             region = NULL,
-             key = spaces_key,
-             secret = spaces_secret,
-             base_url = spaces_base,
-             ...)
+                    region = NULL,
+                    key = spaces_key,
+                    secret = spaces_secret,
+                    base_url = spaces_base,
+                    ...)
 
   if (res) message(sprintf("New space %s created successfully", name))
 }
+
+
+
+#' Retrieve an Object from a Space
+#'
+#' @param object (character) The Object to get
+#' @param space (character) The Space the Object is found in
+#' @param spaces_key (character) String containing a spaces access key. If
+#'   missing, defaults to value stored in an environment variable
+#'   \code{DO_SPACES_ACCESS_KEY}.
+#' @param spaces_secret (character) String containing the secret associated
+#'   with the spaces key. If missing, defaults to value stored in an environment
+#'   variable \code{DO_SPACES_SECRET_KEY}.
+#' @param ... Additional argument passed to \code{\link[aws.s3]{get_object}}
+#'
+#' @return The raw response body of the request
+#' @export
+#'
+#' @references \url{https://developers.digitalocean.com/documentation/spaces/#get-object}
+#'
+#' @examples
+#' \dontrun{
+#' spaces_put_object("./some-file.txt", "my-object", "my-space")
+#' spaces_get_object("my-object", "my-space")
+#' }
+spaces_get_object <- function(object, space, spaces_key = NULL, spaces_secret = NULL, ...) {
+  spaces_key <- check_space_access(spaces_key)
+  spaces_secret <- check_space_secret(spaces_secret)
+
+  aws.s3::get_object(object,
+                     space,
+                     check_region = FALSE,
+                     key = spaces_key,
+                     secret = spaces_secret,
+                     base_url = spaces_base,
+                     ...)
+}
+
+#' Get information about an Object
+#' @param object (character) The Object to get information about
+#' @param space (character) The Space the Object is found in
+#' @param spaces_key (character) String containing a spaces access key. If
+#'   missing, defaults to value stored in an environment variable
+#'   \code{DO_SPACES_ACCESS_KEY}.
+#' @param spaces_secret (character) String containing the secret associated
+#'   with the spaces key. If missing, defaults to value stored in an environment
+#'   variable \code{DO_SPACES_SECRET_KEY}.
+#' @param ... Additional argument passed to \code{\link[aws.s3]{head_object}}
+#'
+#' @return A list of headers associated with the Object
+#' @export
+#'
+#' @references \url{https://developers.digitalocean.com/documentation/spaces/#get-object-info}
+#'
+#' @examples
+#' \dontrun{
+#' spaces_put_object("./some-file.txt", "my-object", "my-space")
+#' spaces_head_object("my-object", "my-space")
+#' }
+spaces_head_object <- function(object, space, spaces_key = NULL, spaces_secret = NULL, ...) {
```

Comment:
Toughts on the output structure from this function?  Maybe make it into a named list instead? just via `attributes(x)`

Reply:
I like that.